### PR TITLE
Moving loss and metrics to model.compile

### PR DESCRIPTION
**Pull request by oliverholworthy**
Updating Example Notebook to match changes made to the loss/metrics in https://github.com/NVIDIA-Merlin/models/pull/431

### CODE REVIEW

File: examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb [hunk 1]
```diff
@@ -391,7 +391,7 @@
     "    embedding_dim=64,\n",
```
Comment (rnyak) [inline examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb:391]: <pre class="ql-syntax" spellcheck="false">Line #4.        loss="categorical_crossentropy", 
</pre><p>Instead of setting loss as a string, it'd be better to demonstrate how  people can set from_logits=False/True arg  and other args in the <code>tf.keras.losses.CategoricalCrossentropy() </code>	? </p><p> </p><p>I'd prefer an example something like that to be similar to tf.keras notations:</p><p> </p><pre class="ql-syntax" spellcheck="false">model.compile(
&nbsp;&nbsp;optimizer='adam',&nbsp;
&nbsp;&nbsp;run_eagerly=False,&nbsp;
&nbsp;&nbsp;loss=tf.keras.losses.CategoricalCrossentropy()
&nbsp;&nbsp;metrics=[mm.RecallAt(10), mm.NDCGAt(10)]
)
</pre><p> </p><p>what do you think?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Merlin/pull/340/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (karlhigley) [reply examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb:391]: I think this is just a change to fix the notebooks after underlying changes in Merlin Models broke them, but it's also worth considering how the examples could be improved once we make the fixes.
